programs/vest_api/src/vest_processor.rs

```diff
@@ -0,0 +1,324 @@
+//! vest program
+use crate::date_instruction::DateConfig;
+use crate::{
+    vest_instruction::{VestError, VestInstruction},
+    vest_state::VestState,
+};
+use bincode::deserialize;
+use chrono::prelude::*;
+use solana_config_api::get_config_data;
+use solana_sdk::{
+    account::{Account, KeyedAccount},
+    instruction::InstructionError,
+    pubkey::Pubkey,
+};
+
+fn parse_date_account(
+    keyed_account: &KeyedAccount,
+    expected_pubkey: &Pubkey,
+) -> Result<Date<Utc>, InstructionError> {
+    if keyed_account.account.owner != solana_config_api::id() {
+        return Err(InstructionError::IncorrectProgramId);
+    }
+
+    if keyed_account.unsigned_key() != expected_pubkey {
+        return Err(VestError::Unauthorized.into());
+    }
+
+    let config_data = get_config_data(&keyed_account.account.data).unwrap();
```

Comment:
this can still fall over with a deserialization failure...

Reply:
In theory? Or do we know how to write a test that reaches that case?

Reply:
pass an account with data.len() of 1 to get_config_data()

Reply:
I don't understand. Also considering dumping the Config program in favor of something simpler. You're saying the Config program can produce an account it can't deserialize? Note the Config id check above.

Reply:
I can construct a config account without ever involving the config program

Reply:
Ah, so the trick is an unintialized config account with not enough space for the config struct? That's something I can write a test for. :)

Reply:
yeap

Reply:
Fixed!